src/unittest/test_moveaction.cpp

```diff
@@ -39,50 +40,39 @@ class TestMoveAction : public TestBase
 	void testMoveSomewhere(ServerActiveObject *obj, IGameDef *gamedef);
 	void testMoveUnallowed(ServerActiveObject *obj, IGameDef *gamedef);
 	void testMovePartial(ServerActiveObject *obj, IGameDef *gamedef);
+
 	void testSwap(ServerActiveObject *obj, IGameDef *gamedef);
 	void testSwapFromUnallowed(ServerActiveObject *obj, IGameDef *gamedef);
 	void testSwapToUnallowed(ServerActiveObject *obj, IGameDef *gamedef);
+
+	void testCallbacks(ServerActiveObject *obj, Server *server);
+	void testCallbacksSwap(ServerActiveObject *obj, Server *server);
 };
 
 static TestMoveAction g_test_instance;
 
-const static char *helper_lua_src = R"(
-core.register_allow_player_inventory_action(function(player, action, inventory, info)
-	if action == "move" then
-		return info.count
-	end
-
-	if info.stack:get_name() == "default:water" then
-		return 0
-	end
-	if info.stack:get_name() == "default:lava" then
-		return 5
-	end
-
-	return info.stack:get_count()
-end)
-)";
-
 void TestMoveAction::runTests(IGameDef *gamedef)
 {
 	MockServer server(getTestTempDirectory());
 
-	const auto helper_lua = getTestTempFile();
-	std::ofstream ofs(helper_lua, std::ios::out | std::ios::binary);
-	ofs << helper_lua_src;
-	ofs.close();
-
 	ServerScripting server_scripting(&server);
 	try {
-		server_scripting.loadMod(Server::getBuiltinLuaPath() + DIR_DELIM "init.lua", BUILTIN_MOD_NAME);
-		server_scripting.loadMod(helper_lua, BUILTIN_MOD_NAME);
+		// FIXME: When removing the line below, the unittest does NOT crash
+		// (but it should) when running all unittests in order or registration.
+		server.m_modmgr = std::make_unique<ServerModManager>(server.m_path_world);
```

Comment:
what's the mod manager used for?

Reply:
`core.get_modpath` (and similar) rely on the existence of `ServerModManager`. When I last checked, the pointer was not `nullptr`, indicating a deeper issue. However, that could also have changed since the last rebase.